tests/view/RoomsTest.java

```diff
@@ -0,0 +1,178 @@
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            null,
+            null);
+
+    RoomsState testState = buildTestState();
+    testState.setRoomToCreateName("Test Room!");
+    roomsViewModel.setState(testState);
+
+    JButton createRoomButton = roomsView.getCreateRoomButton();
+    createRoomButton.doClick();
+
+    String regex = "^Authentication failed: .*";
+
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(roomsViewModel.getState().getError());
+    Assert.assertTrue(matcher.find());
+  }
+
+  @Test
+  public void testSendMessage() {
+    OkHttpClient client = new OkHttpClient();
+    ViewManagerModel viewManagerModel = new ViewManagerModel();
+
+    SearchViewModel searchViewModel = new SearchViewModel();
+    SearchedViewModel searchedViewModel = new SearchedViewModel();
+    SearchDataAccessInterface searchDataAccessObject = new ElasticsearchDataAccessObject(client);
+    SearchController searchController =
+        SearchUseCaseFactory.createSearchController(
+            searchViewModel, searchDataAccessObject, viewManagerModel, searchedViewModel);
+
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            searchController,
+            null);
+
+    RoomsState testState = buildTestState();
+    testState.setRoomUid(testState.getAvailableRooms().get(0).getUid());
+    testState.setSendMessage("Test message!");
+    System.out.println(testState.getAvailableRooms().get(0).getMessages());
+    roomsViewModel.setState(testState);
+
+    JButton sendMessageButton = roomsView.getSendMessageButton();
+    sendMessageButton.doClick();
+
+    System.out.println(roomsViewModel.getState().getSuccess());
+
+    Assert.assertTrue(roomsViewModel.getState().getSuccess().equals("Success"));
+  }
+
+  @Test
+  public void testAddUserToRoom() {
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    OkHttpClient client = new OkHttpClient();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            null,
+            null);
+
+    RoomsState testState = buildTestState();
+    testState.setUserToAddEmail(createDummyDisplayUser().getEmail());
+    roomsViewModel.setState(testState);
+
+    JButton addUserButton = roomsView.getAddUserButton();
+    addUserButton.doClick();
+
+    String regex =
+        "Unable to add \\s*([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,})\\s*to room:";
```

Comment:
why use a complicated regex if you're not going to go [all the way](https://emailregex.com/)